core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -825,6 +826,11 @@ class SparkContext(config: SparkConf) extends Logging {
   def getLocalProperty(key: String): String =
     Option(localProperties.get).map(_.getProperty(key)).orNull
 
+  /** Set the UUID of the Spark session that starts the current job. */
+  def setSparkSessionUUID(uuid: String): Unit = {
```

Comment:
I would prefer to use the same approach with `JobArtifactSet`. We shouldn't add the concept of SQL/SparkSession here. You can hold a default key here, and set the state within `SparkSession`, see also https://github.com/apache/spark/commit/57bbb4c6e7309e0beaeb3618978dcaf75ce9a7fe#diff-1e596e9e9bd50e3075847a66f8fecee741289b2b0478f9099c3bf2b680ebaa35R1741